app/forms/waste_carriers_engine/service_provided_form.rb

```diff
@@ -4,9 +4,11 @@ module WasteCarriersEngine
   class ServiceProvidedForm < BaseForm
     attr_accessor :is_main_service
 
+    validates :is_main_service, "waste_carriers_engine/yes_no": true
+
     def initialize(transient_registration)
       super
-      self.is_main_service = @transient_registration.is_main_service
+      self.is_main_service = transient_registration.is_main_service
```

Comment:
NIT Adding a line between super and this as per other changes

```suggestion
      
      self.is_main_service = transient_registration.is_main_service
```